### DISCUSSION TITLE
Maintaining temperature in shear flow in HOOMD

### DISCUSSION THREAD

**Discussion by mubisk** [glotzerlab/hoomd-blue#1701] [Support] [answered]
Dear all,

I am trying to shear a system of ring polymers using **hoomd.update.BoxResize** where, the **hoomd.box.xy** is constantly updated and **md.methods.Langevin** is used. However, while shearing, the temperature is increasing quite fast. I tried to change the value of alpha in the Langevin method command but its not enough to maintain a temperature of kBT=0.1.  Can anyone please help me?
Also, in the loggable quantities >> Kinetic temperature.......... does this quantity displays only temperature or the thermal energy kT?

**Comment by mubisk** [comment]
Hi, I got the error. I had set alpha=0 first and then defined gamma as follows:
langevin = md.methods.Langevin(filter=hoomd.filter.All(), alpha=0.0)
langevin.gamma.default = 260.0

Then I deleted alpha value and defined it again as follows:
langevin = md.methods.Langevin(filter=hoomd.filter.All(), kT=temp,tally_reservoir_energy=True)
langevin.gamma.default = 260.0

and now its working and keeping the temperature ~0.1.
Though I am quite sure about the kinetic temperature in the loggable quantities must be kT but would liek to confirm from you all once that its kT (thermal energy) and not just the kinetic temperature as the name suggests!!

**Reply by joaander** [reply]
The equation used to compute `kinetic_temperature` is [clearly documented](https://hoomd-blue.readthedocs.io/en/v4.4.1/module-md-compute.html#hoomd.md.compute.ThermodynamicQuantities.kinetic_temperature).

**Reply by mubisk** [reply]
Hi. Yeah but I would like to know the term kinetic temperature displays kT (thermal energy with the multiplication factor of boltzmann constant k) or it displays only T. generally in our simulations we express temperature in terms of energy. I was confused by the name kinetic temperature whether it displays temperature (without k factor) or the energy values in the loggable quantities??

**Reply by joaander** [reply, marked as answer]
As mentioned in the documentation I linked to, `kinetic_temperature` is kT in energy units.
